diesel_codegen/src/associations.rs

```diff
@@ -0,0 +1,103 @@
+use syn;
+use quote;
+
+use model::{Model, infer_association_name};
+use util::str_value_of_meta_item;
+
+pub fn derive_associations(input: syn::MacroInput) -> quote::Tokens {
+    let mut derived_associations = Vec::new();
+    let model = t!(Model::from_item(&input, "Associations"));
+
+    for attr in &input.attrs {
+        if attr.name() == "has_many" {
```

Comment:
I avoided that because of the additional rightward drift, and the need for a no-op `_` arm.